### PR TITLE
add a "flag" filter to registry 

### PR DESCRIPTION
**Pull request by taylorhelene**
I have added a flag button on the search that adds flags like 'deprecated', 'native' and 'first party'. I have changed these thee components:
   - assets/js/registrySearch.js
   - layouts/partials/ecosystem/registry/entry.html
   - layouts/shortcodes/ecosystem/registry/search-form.html

   
flags

deprecated

### CODE REVIEW

File: content/es/docs/concepts/observability-primer.md
Comment (svrnm) [inline content/es/docs/concepts/observability-primer.md:1]: this shouldn't be part of this PR

Comment (taylorhelene) [reply content/es/docs/concepts/observability-primer.md:1]: I have removed the file

**Comment by taylorhelene** [comment]
It works when removing spaces from the "first party" flag

File: layouts/shortcodes/ecosystem/registry/search-form.html [hunk 1]
```diff
@@ -88,6 +124,16 @@
         {{ end -}}
       </div>
 
+      <!-- Flags Filter Dropdown -->
+    <button class="btn btn-outline-secondary dropdown-toggle" id="flagsDropdown" type="button" 
+    data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false">Flags</button>
+    <div class="dropdown-menu" id="flagsFilter">
+    <a value="all" class="dropdown-item">Any Flag</a>
+    {{ range $flag := $uniqueFlags -}}
+    <a value="{{ $flag }}" id="flag-item-{{ $flag }}" class="dropdown-item">{{ humanize $flag }}</a>
```
Comment (svrnm) [inline layouts/shortcodes/ecosystem/registry/search-form.html:133]: Screenshot 2024-10-08 at 11 31 45

In the text visible to the user you will need to keep the space between "first" and "party". A way to solve this is by using "first-party" instead of "firstparty", since the humanize will take care of turning that into a space, see https://gohugo.io/functions/inflect/humanize/

**Comment by taylorhelene** [comment]
I have added "first_party" then removed the console logs and prints I had added.

File: assets/js/registrySearch.js [hunk 2]
```diff
@@ -38,6 +38,8 @@ let pathName = window.location.pathname;
 let searchQuery = '';
 let selectedLanguage = 'all';
 let selectedComponent = 'all';
+let selectedFlag = 'all'; // Added selectedFlag 
+
```
Comment (svrnm) [inline assets/js/registrySearch.js:42]:
```suggestion
```

File: assets/js/registrySearch.js [hunk 3]
```diff
@@ -49,22 +51,24 @@ if (pathName.includes('registry')) {
...
+        document.getElementById(`component-item-${selectedComponent}`).textContent;
+    }
+    if (selectedFlag !== 'all') {
+      document.getElementById('flagsDropdown').textContent =
+        document.getElementById(`flag-item-${selectedFlag}`).textContent;
     }
     updateFilters();
   }
 
+
```
Comment (svrnm) [inline assets/js/registrySearch.js:71]:
```suggestion
```

File: assets/js/registrySearch.js [hunk 4]
```diff
@@ -208,34 +229,44 @@ function setInput(key, value) {
   history.replaceState(null, null, '?' + queryParams.toString());
 }
 
-// Filters items based on language and component filters
+// Filters items based on language, component, and flags
```
Comment (svrnm) [inline assets/js/registrySearch.js:232]:
```suggestion
// Filters items based on language, component and flags
```

File: assets/js/registrySearch.js [hunk 5]
```diff
@@ -208,34 +229,44 @@ function setInput(key, value) {
...
-        (dc === selectedComponent || selectedComponent === 'all') &&
-        (dl === selectedLanguage || selectedLanguage === 'all')
-      ) {
+      const df = element.dataset.registryflags ? element.dataset.registryflags.split(' ').map(f => f.toLowerCase()) : [];
+
+      const componentMatches = (dc === selectedComponent || selectedComponent === 'all');
+      const languageMatches = (dl === selectedLanguage || selectedLanguage === 'all');
+      const flagMatches = (selectedFlag === 'all' || df.includes(selectedFlag.toLowerCase()));
+
+
```
Comment (svrnm) [inline assets/js/registrySearch.js:249]:
```suggestion

```

File: assets/js/registrySearch.js [hunk 6]
```diff
@@ -208,34 +229,44 @@ function setInput(key, value) {
...
+      const componentMatches = (dc === selectedComponent || selectedComponent === 'all');
+      const languageMatches = (dl === selectedLanguage || selectedLanguage === 'all');
+      const flagMatches = (selectedFlag === 'all' || df.includes(selectedFlag.toLowerCase()));
+
+      
+      if (flagMatches) {
+        console.log('Flag matches:', df);
+      }
+
+
```
Comment (svrnm) [inline assets/js/registrySearch.js:254]:
```suggestion

```

File: layouts/shortcodes/ecosystem/registry/search-form.html [hunk 7]
```diff
@@ -34,6 +34,39 @@
...
+    {{ $flags = $flags | append "native" -}}
+  {{ end -}}
+  {{ if .isFirstParty -}}
+    {{ $flags = $flags | append "first_party" -}}
+  {{ end -}}
+  {{ if .isNew -}}
+    {{ $flags = $flags | append "new" -}}
+  {{ end -}}
+  {{ if .usedInDemo -}}
+    {{ $flags = $flags | append "used in demo" -}}
```
Comment (svrnm) [inline layouts/shortcodes/ecosystem/registry/search-form.html:51]:
```suggestion
    {{ $flags = $flags | append "used_in_demo" -}}
```

Although currently not used, here you also might want to use _ instead of spaces

**Review by svrnm — CHANGES_REQUESTED** [review]
Looks great overall, some minor change requests.

**Comment by svrnm** [comment]
lgtm overall, can you do the following please to fix the remaining CI issues:

* update your branch with the latest commit to `main`
* run `npm run fix:all` locally and commit the changes.

**Comment by taylorhelene** [comment]
Do I sync my fork with the latest commits from main or do I push to origin main. I do not understand the first suggestion of updating branch with the latest commit to main.

**Comment by svrnm** [comment]
Thank you @taylorhelene! This looks really good, I consider this as done! After the outreachy application phase we will take another look and see if we can merge this PR.

**Comment by taylorhelene** [comment]
Okay then.  Thank you for the directions given.

**Review by svrnm — APPROVED** [review]

**Comment by svrnm** [comment]
Thank you once again @taylorhelene !

File: content/es/docs/concepts/observability-primer.md
Comment (chalin) [inline content/es/docs/concepts/observability-primer.md:1]: Why is this PR deleting an `es` page? 

/cc @svrnm @theletterf

**Comment by chalin** [comment]
Also, note the following bug. I'm not sure if it was introduced by this PR or not:

- #5588

File: content/es/docs/concepts/observability-primer.md
Comment (svrnm) [reply content/es/docs/concepts/observability-primer.md:1]: Oh wow. I missed that. Huge apologies!!
